zebROS_ws/src/tf_object_detection/src/object_detection.cpp

```diff
@@ -143,6 +400,38 @@ void callback(const field_obj::TFDetectionConstPtr &objDetectionMsg, const senso
 		}
 	}
 }
+
+
+// A callback function for testing usefulDepthMat. This is for a subscriber that
+// subscribes to the `test_depth` topic.
+void testUsefulDepthMatCallback(const std_msgs::String::ConstPtr& msg) {
```

Comment:
An idea for how to make this more ROS-like. Instead of a separate subscriber specifically for test data, perhaps move the guts of this code to another node. It would read the data in and format it into the depth image + bounding boxes format expected by the main subscriber and publish those.  
That would end up testing more of the full node behavior, all the way from receiving a depth image plus detection code.

Reply:
I was mainly just using that function so I could test depth data images that I made (but weren't taken by the robot) with the algorithms and was able to test the full node behavior with some of the rosbags of the power cells, so is it okay if I leave it in the code (or move it to a separate node with only that function)?

Reply:
I think it would be a good exercise to move it to a separate node like I mentioned. We don't have a lot of stand-alone tests for our code, so whatever you learn from doing so would be helpful in the future as we try to be a bit more test-driven, especially in cases like this where we can easily add in real data we collect in the future.